src/Build/Logging/BinaryLogger/BinaryLogReplayEventSource.cs

```diff
@@ -27,9 +69,12 @@ static BinaryLogReplayEventSource()
         }
 
         /// <summary>
-        /// Raised once <see cref="BuildEventArgsReader"/> is created during replaying
+        /// Unknown build events or unknown parts of known build events will be ignored if this is set to true.
         /// </summary>
-        public event Action<IBuildEventArgsReaderNotifications>? NotificationsSourceCreated;
+        public bool AllowForwardCompatibility { private get; init; }
```

Comment:
I would make forward compat the default. Most libraries want to just read the most data and if they want to be meticulous they can opt-in to strictness.

Reply:
I wanted it as well - but team consesus was to keep it opt-in as it is technically a breaking change. That point of view makes a sense - especially in the initial phase.

How about having it dafault (opt-out) in the viewer and binlogtool (as those are actuall toolings rather then api), opt-in in the first roll-out in MSBuild API and if everything goes smooth the next preview version we flip to opt-out in MSBuild?

If that doesn't sound as bad idea I'd create an item for this and push through planning. We'd as well need to have sensible default errors handler (likely the synthetic build errors you suggested in the viewer PR).

Reply:
Sounds fine. We're only talking about replaying binlogs through msbuild.exe here, right? It's not a common scenario I think.